### ISSUE TITLE
Highlight undefined steps in Scenario Outlines

### ISSUE THREAD

**Issue by kiran-epic** [cucumber/vscode#90]
When using scenario outline for data-driving with the help of examples, The following issues occur :

- Only the parameters which are present in the Examples table are highlighted. For example in the below image, the "EducatorLogin" & "Dashboard" parameters are not highlighted as it's mentioned directly in the step as they don't change so no need to put it in the Examples table
![image](https://user-images.githubusercontent.com/98785077/182957901-0fe87b5e-9de1-46e9-b4d5-5de4f60cc83c.png)

- The missing step highlighter/generator doesn't run if there is an example table in the scenario. For example in the below image, the step `Given QWERTY` is invalid, but it's not highlighted & pressing Cmd+. doesn't give to generate a step definition for it.
![image](https://user-images.githubusercontent.com/98785077/182958055-9406092c-ccfb-4c38-845f-561189fe4d8e.png)

But if the example table is removed, it starts highlighting the step & also suggests action to generate step definition
![image](https://user-images.githubusercontent.com/98785077/182957687-85708130-7995-408d-aa96-a25ef19a504b.png)
![image](https://user-images.githubusercontent.com/98785077/182957466-c079ba71-f264-4aff-b648-da63526c64f9.png)

**Comment by aslakhellesoy** [comment]
This is actually by design.

Highlighting of parameters is done by matching steps against step definitions. For steps in regular scenarios we can do this. For steps in scenario outlines it is tricky to do, because each outline step becomes multiple steps after the `<>` is substituted.

Consider this example:

```gherkin
Scenario Outline:
  Given I have <count> cukes in my <container>
  
  Examples:
    | count | container |
    |     4 | basket    |
    |     7 | bag       |
```

This becomes two scenarios, each with the following step:

* `Given I have 4 cukes in my basket`
* `Given I have 7 cukes in my bag`

Now, imagine we have two step definitions with the following expressions:

* `I have 4 cukes in my {word}`
* `I have {int} cukes in my bag`

The first step would match the first expression, and the second step would match the second expression. Each match has only a single parameter.

How should we highlight in this case?

I suppose a compromise could be to highlight using matches if there are no `<>` placeholders, and highlight the `<>` placeholders otherwise.

**Comment by aslakhellesoy** [comment]
@kiran-epic any feedback on my proposal?

**Comment by kiran-epic** [comment]
@aslakhellesoy That sounds good to me!

**Comment by kiran-epic** [comment]
@aslakhellesoy Would you also fix the bug where the autocomplete & missing step highlighter stops working when using Examples ?

**Comment by aslakhellesoy** [comment]
I haven’t offered to fix anything, just discussing a potential solution.

This project relies on contributions from our community.

**Comment by cgarzon11** [comment]
Hi! I have the same problems as @kiran-epic working with scenario outlines.

**Comment by kieran-ryan** [comment]
Just encountered this issue looking through the backlog.

I had captured a feature request for this support in the Cucumber Language Service - where the changes would be required - with some possible considerations and options (https://github.com/cucumber/language-service/issues/149). If you have any feedback, that would be great - to help define the neccessary changes.

Great to see the discussion and complexities captured here - interesting challenge to resolve!

**Comment by Alphapage** [comment]
Same problem for me, but even with Scenario : not only Scenario Outline.
A workaround is to comment the Examples table, but latest version 1.8.0 removes the comments on save !!!
It would be great to generate at least the first Step definition to remove the highlighting when Outline.
Examples seems to be the problem to me and must be ignored by the extension for syntax highlighting...
The emergency problem is to be able to keep comments on save.

**Comment by kieran-ryan** [comment]
@Alphapage, deletion of trailing comments is resolved by cucumber/gherkin-utils#38 and will be available following a gherkin-utils release of [merged changes](https://github.com/cucumber/gherkin-utils/blob/0603bff54cc1c2e23c206c257d05fbd9baf4ae71/CHANGELOG.md) and thereafter of a Visual Studio Code extension release.

**Comment by Alphapage** [comment]
Sorry, I read it was fixed in v.1.8.0 somewhere or I misunderstood something...
Thank you anyway for the clarification and I hope the extension will be updated soon +1
